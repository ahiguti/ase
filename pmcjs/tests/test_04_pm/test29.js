
use('Perl');
Perl.Use('Unicode::UCD');
var ucd = Perl.Functions['Unicode::UCD'];
var str = 'Aあアぁｱ漢';
for (var i = 0; i < str.length; ++i) {
  var c = str.charAt(i);
  var code = str.charCodeAt(i);
  var ci = ucd.charinfo(code);
  print("char: " + c);
  print("code: " + code.toString(16));
  print("name: " + ci.name);
}

