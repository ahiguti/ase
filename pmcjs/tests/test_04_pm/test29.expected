char: A
code: 41
name: LATIN CAPITAL LETTER A
char: あ
code: 3042
name: HIRAGANA LETTER A
char: ア
code: 30a2
name: KATAKANA LETTER A
char: ぁ
code: 3041
name: HIRAGANA LETTER SMALL A
char: ｱ
code: ff71
name: HALFWIDTH KATAKANA LETTER A
char: 漢
code: 6f22
name: CJK UNIFIED IDEOGRAPH-6F22
